docs/iris/src/userguide/real_and_lazy_data.rst

```diff
@@ -1,246 +1,211 @@
+:meth:`~iris.cube.Cube.has_lazy_data`. For example::
 
-    >>> import iris
     >>> cube = iris.load_cube(filename, 'air_temp.pp')
     >>> cube.has_lazy_data()
     True
     >>> cube.data
     >>> cube.has_lazy_data()
     False
 
-When does my Data Become Real?
+When does my data become real?
 ------------------------------
 
-If the data on your cube is in its lazy state, it will only become real if you
-'touch' the data.  This means any way of directly accessing the data, such as
-assigning it to a variable or simply using 'cube.data' as in the example above.
-
-Any action which requires the use of actual data values (such as cube maths)
-will also cause the data to be loaded into memory, although data realization
-is always deferred until the result is requested:
-
-.. doctest::
+Most operations on data arrays can be run equivalently on both real and lazy data.
+If the data array is real then the operation will be run on the data array
+immediately with the results becoming available as soon as processing is completed.
+If the data array is lazy then the operation will be deferred until you request
+the result (such as when you call ``cube.data``). In this case the data array will
+remain lazy::
 
     >>> cube = iris.load_cube(iris.sample_data_path('air_temp.pp'))
     >>> cube.has_lazy_data()
     True
     >>> cube += 5
     >>> cube.has_lazy_data()
     True
-    >>> cube.data
-    >>> cube.has_lazy_data()
-    False
 
-You can also convert realized data back into a lazy array:
+This is referred to as **lazy evaluation**.
 
-.. doctest::
+Certain operations, including regridding and plotting, can only be run on real data.
+Calling such operations on lazy data will automatically realise your lazy data.
 
-    >>> cube.has_lazy_data()
-    False
-    >>> cube.data = cube.lazy_data()
-    >>> cube.has_lazy_data()
-    True
+You can also realise (and so load into memory) your cube's lazy if you 'touch' the data.
+This means directly accessing the data by calling ``cube.data``, as in the previous example.
 
-Core data refers to the current state of the cube's data, be it real or
-lazy.  This can be used if you wish to refer to the data array but are
-indifferent to its current state.  If the cube's data is lazy, it will not be
-realized when you reference the core data attribute (?):
+Core data
+^^^^^^^^^
 
-.. doctest::
+Cubes have the concept of "core data". This returns the cube's data in its
+current state. If a cube has lazy data, calling the cube's
+:meth:`~iris.cube.Cube.core_data` method will return the cube's lazy dask array.
+If the cube has real data, calling the cube's
+:meth:`~iris.cube.Cube.core_data` method will return the cube's real NumPy array.
+
+Calling the cube's :meth:`~iris.cube.Cube.core_data` method will not change the
+state of the cube's data.  Thus, if the cube's data is lazy then calling
+:meth:`~iris.cube.Cube.core_data` will return the cube's lazy data and not
+realise it.
```

Comment:
I am not sure if we need this extra bit.  It seems to be restating the sentence on L.86/87.